lib/code-manager.coffee

```diff
@@ -81,12 +81,13 @@ class CodeManager
 
 
     normalizeString: (code) ->
-        return code.replace /\r\n|\r/g, '\n'
+        if code?
+            return code.replace /\r\n|\r/g, '\n'
 
 
     getFoldRange: (row) ->
         range = @editor.languageMode.rowRangeForCodeFoldAtBufferRow(row)
-        if @getRow(range[1] + 1).trim() is 'end'
+        if @getRow(range[1] + 1)?.trim() is 'end'
```

Comment:
You can really apply the `.trim()` after the `?` operator in coffeescript? Crazy.